AI #2177: Revise content of the README file (#2323)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: README.md

## Overview

The FinOps Open Cost and Usage Specification (FOCUS) is a community-driven, open specification that defines a common schema for technology cost and usage data across cloud, SaaS, data center, and other technology categories.

FOCUS establishes a consistent, vendor-neutral vocabulary for billing and usage data. It defines a collection of standardized datasets, specifying the columns (dimensions and metrics), their associated requirements, and specification-wide attributes needed to enable interoperable, comparable, and analysis-ready data across providers and technology categories.

The project is actively maintained and adopted by a growing ecosystem of cloud providers, SaaS vendors, enterprises, and FinOps tooling platforms. This repository contains the specification releases, source, build and validation tooling, and supporting contributor guidance that help practitioners, platforms, and providers produce and consume data that is consistent, portable, and aligned to FinOps practices.

> 

Author: ```suggestion

### Project Naming

* The official project name is **FinOps Open Cost and Usage Specification**
* The official acronym is **FOCUS**
* The term **FOCUS Specification** is the accepted shorthand for the project's output

```

Implementations should explicitly declare the version of FOCUS they align to.

For release process details and version planning context, see [RELEASE-PLANNING.md](RELEASE-PLANNING.md).

---

## About the Project

FOCUS is developed under an open governance model with participation from providers, vendors, and practitioners across the ecosystem.

For additional details, see the project repository:

* [Project Charter](https://github.com/FinOps-Open-Cost-and-Usage-Spec/foundation/blob/main/FOCUS_-_Membership_Agreement_Package_for_use.pdf)
* [Operating Procedures](https://github.com/FinOps-Open-Cost-and-Usage-Spec/foundation/blob/main/operating_procedures.md)
* [Steering Committee](https://github.com/FinOps-Open-Cost-and-Usage-Spec/foundation/blob/main/steering_committee.md)
* [Release Planning](RELEASE-PLANNING.md)
* [Contribution Process](https://github.com/FinOps-Open-Cost-and-Usage-Spec/foundation/blob/main/contributing.md)

---

## Project Naming

* The official project name is **FinOps Open Cost and Usage Specification**
* The official acronym is **FOCUS**

> * The term **FOCUS Specification** is the accepted shorthand for the project's output

Author: ```suggestion

```

## Overview

> The FinOps Open Cost and Usage Specification (FOCUS) is a community-driven, open specification that defines a common schema for technology cost and usage data across cloud, SaaS, data center, and other technology categories.

Reviewer: Are there any FinOps Foundation materials we should link here?  Or do you prefer keeping it more separate from F2?

> The FinOps Open Cost and Usage Specification (FOCUS) is a community-driven, open specification that defines a common schema for technology cost and usage data across cloud, SaaS, data center, and other technology categories.

Author: ```suggestion
The FinOps Open Cost and Usage Specification (FOCUS) is a community-driven, open specification that defines a common schema for technology cost and usage data across cloud, SaaS, data center, and other [technology categories](https://www.finops.org/framework/technology-categories/).
```

> FOCUS establishes a consistent, vendor-neutral vocabulary for billing and usage data. It defines a set of columns (dimensions and metrics), column-level requirements, and specification-wide attributes that enable interoperable, comparable, and analysis-ready datasets across providers and technology categories.

Reviewer: I'd like to make clear here that it's a collection of datasets.  Feel free to wordsmith further -- mentioning "datasets" is the main point.

```suggestion
FOCUS establishes a consistent, vendor-neutral vocabulary for billing and usage data. It defines a collection of standardized datasets, specifying the columns (dimensions and metrics), their associated requirements, and specification-wide attributes needed to enable interoperable, comparable, and analysis-ready data across providers and technology categories.
```

The project is actively maintained and adopted by a growing ecosystem of cloud providers, SaaS vendors, enterprises, and FinOps tooling platforms. This repository contains the specification releases, source, build and validation tooling, and supporting contributor guidance that help practitioners, platforms, and providers produce and consume data that is consistent, portable, and aligned to FinOps practices.

### Project Naming

* The official project name is **FinOps Open Cost and Usage Specification**
* The official acronym is **FOCUS**
* The term **FOCUS Specification** is the accepted shorthand for the project's output

For a deeper overview of FOCUS, its capabilities, and adoption, see the [FOCUS website](https://focus.finops.org).

## Accessing the Specification

Use the following links to access the latest specification artifacts:

* [Latest Public Release](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/releases/latest): Stable release for general implementation and reference.
* [Latest Working Draft](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/releases/tag/latest-draft): In-progress draft reflecting current working-group development.

> 

Reviewer 2: ```suggestion
* [Latest Working Draft](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/releases/tag/latest-draft): In-progress draft reflecting current working group development.

```

Two small fixes combined:
1. "working-group": FOCUS consistently uses "working group" as two unhyphenated words throughout the spec and its guidelines. The hyphenated form could imply an official entity name.
2. Double blank line: Lines 25-26 are both blank before `## Contributor Getting Started`. The rest of the file uses either a single blank line or a `---` divider — not a double blank.

For topic-specific standards, see:

* [guidelines/contributors/editorial-guidelines.md](guidelines/contributors/editorial-guidelines.md)
* [guidelines/contributors/normative-requirements-guidelines.md](guidelines/contributors/normative-requirements-guidelines.md)
* [guidelines/contributors/spec-change-guidelines.md](guidelines/contributors/spec-change-guidelines.md)

> * Additional project guidance: [guidelines/](guidelines/)

Reviewer: Let's make this list a little less scary by including friendly names:

> * [Contribution Guide](CONTRIBUTING.md)
> * [Development Processes](guidelines/contributors/development-processes.md)
> 
> For topic-specific standards, see:
> 
> * [Editorial Guidelines](guidelines/contributors/editorial-guidelines.md)
> * [Normative Requirements Guidelines](guidelines/contributors/normative-requirements-guidelines.md)
> * [Specification Change Guidelines](guidelines/contributors/spec-change-guidelines.md)
> * See the [Guidelines Directory](guidelines/) for additional project guidance
> 

```suggestion
* [Contribution Guide](CONTRIBUTING.md)
* [Development Processes](guidelines/contributors/development-processes.md)

For topic-specific standards, see:

* [Editorial Guidelines](guidelines/contributors/editorial-guidelines.md)
* [Normative Requirements Guidelines](guidelines/contributors/normative-requirements-guidelines.md)
* [Specification Change Guidelines](guidelines/contributors/spec-change-guidelines.md)
* See the [Guidelines Directory](guidelines/) for additional project guidance
```

---

## Normative Language and Conformance

The keywords **MUST**, **MUST NOT**, **SHOULD**, **SHOULD NOT**, and **MAY** in the specification are to be interpreted as described in BCP 14 (RFC2119 and RFC8174), when and only when they appear in all capitals.

> FOCUS restricts normative language to this subset as defined in the project’s  

Reviewer 3: ```suggestion
FOCUS restricts normative language to this subset as defined in the project's  
```

Smart apostrophe (U+2019) replaced with a straight apostrophe to clear the MD990 linter rule. Trailing two-space soft break preserved.

> [Normative Requirements Guidelines](guidelines/contributors/normative-requirements-guidelines.md).

Reviewer: Strange that this is on two lines...

```suggestion
FOCUS restricts normative language to this subset as defined in the project's [Normative Requirements Guidelines](guidelines/contributors/normative-requirements-guidelines.md).
```

> An implementation is considered **conformant** if it satisfies all **MUST** and **MUST NOT** requirements defined in the specification. Failure to meet any of these requirements results in non-conformance.

Reviewer: @Matt-Cowsert Are you comfortable with this language in light of the conformance program?

I wonder if we should link out to what conformance means to the appropriate place.

Author: Ahh yeah good call. I will let @Matt-Cowsert decide what this should say.

Reviewer 3: Agreed. The current wording is too binary for how the conformance program actually works. Per the FinOps Certified FOCUS Conformant program, individual unmet MUST/MUST NOT requirements are recorded as deviations evaluated against per-dataset allowances and disclosed in a public Gap Report, not treated as blanket non-conformance.

Reframing line 44 to reflect that, plus a stable link to the certifications hub:

```suggestion
Implementations are expected to satisfy the **MUST** and **MUST NOT** requirements defined in the specification. Individual unmet requirements are recorded as deviations and evaluated against per-dataset allowances by the FinOps Certified FOCUS Conformant program for data generators.

For details on certification, see [FinOps Certifications for Organizations](https://www.finops.org/certification-for-organizations/).
```

Pointed to the top-level certifications page rather than a version-specific URL so this doesn't need to be updated each FOCUS release.

The keywords **MUST**, **MUST NOT**, **SHOULD**, **SHOULD NOT**, and **MAY** in the specification are to be interpreted as described in BCP 14 (RFC2119 and RFC8174), when and only when they appear in all capitals.

FOCUS restricts normative language to this subset as defined in the project's [Normative Requirements Guidelines](guidelines/contributors/normative-requirements-guidelines.md).

Implementations are expected to satisfy the **MUST** and **MUST NOT** requirements defined in the specification. Individual unmet requirements are recorded as deviations and evaluated against per-dataset allowances by the FinOps Certified FOCUS Conformant program for data generators.

For details on certification, see [FinOps Certifications for Organizations](https://www.finops.org/certification-for-organizations/).

---

## Versioning the Specification

Changes to the specification are documented in [CHANGELOG.md](CHANGELOG.md).

> FOCUS uses a versioning approach inspired by [Semantic Versioning 2.0](https://semver.org/spec/v2.0.0.html), but versions should not be interpreted as strict compatibility guarantees. New versions primarily introduce additive capabilities, clarifications, and expanded coverage while avoiding breaking changes.

Reviewer 3: The current wording understates how the project handles incompatible changes. The CHANGELOG explicitly classifies changes as Compatible / Migration Compatible / Incompatible, and v1.4 includes the planned removal of ProviderName and PublisherName as an Incompatible Change. Suggesting a rewrite that preserves the additive-first framing while accurately reflecting the deprecation cycle.

```suggestion
FOCUS uses a versioning approach inspired by [Semantic Versioning 2.0](https://semver.org/spec/v2.0.0.html), but versions should not be interpreted as strict compatibility guarantees. New versions favor additive capabilities, clarifications, and expanded coverage. Breaking changes follow a formal deprecation cycle documented in the [CHANGELOG](CHANGELOG.md) and [Specification Change Guidelines](guidelines/contributors/spec-change-guidelines.md).
```

## About the Project

FOCUS is developed under an open governance model with participation from providers, vendors, and practitioners across the ecosystem.

For additional details, see the project repository:

* [Project Charter](https://github.com/FinOps-Open-Cost-and-Usage-Spec/foundation/blob/main/FOCUS_-_Membership_Agreement_Package_for_use.pdf)
* [Operating Procedures](https://github.com/FinOps-Open-Cost-and-Usage-Spec/foundation/blob/main/operating_procedures.md)
* [Steering Committee](https://github.com/FinOps-Open-Cost-and-Usage-Spec/foundation/blob/main/steering_committee.md)
* [Release Planning](RELEASE-PLANNING.md)
* [Contribution Process](https://github.com/FinOps-Open-Cost-and-Usage-Spec/foundation/blob/main/contributing.md)

---

> ## Project Naming

Reviewer: Should we maybe fold this into "About the Project"?  Would be good to end on the links -- this section feels tacked on a bit.

* The official project name is **FinOps Open Cost and Usage Specification**
* The official acronym is **FOCUS**

> * The term **FOCUS Specification** is the accepted shorthand for the project's output

Reviewer 3: ```suggestion
* The term **FOCUS Specification** is the accepted shorthand for the project's output
```

Same bullet text. Accepting the suggestion lets GitHub re-commit the file with a trailing newline (the diff shows `\ No newline at end of file`).

> * The term **FOCUS Specification** is the accepted shorthand for the project's output

Author: ```suggestion
* The term **FOCUS Specification** is the accepted shorthand for the project's output

```

> * [Contribution Process](https://github.com/FinOps-Open-Cost-and-Usage-Spec/foundation/blob/main/contributing.md)

Reviewer 3: The README links to two contribution-related docs: this one and `CONTRIBUTING.md` (line 31, "Contribution Guide"). They cover overlapping ground, which can confuse new contributors deciding which to follow. Relabeling this entry to clarify it points to the foundation-level baseline. (The repo's CONTRIBUTING.md remains the canonical onboarding doc.)

```suggestion
* [Foundation Contribution Process](https://github.com/FinOps-Open-Cost-and-Usage-Spec/foundation/blob/main/contributing.md)
```
